src/TrigScint/QIEInputPulse.cxx

```diff
@@ -0,0 +1,183 @@
+/**
+ * @file QIEInputPulse.cxx
+ * @author Niramay Gogate, Texas Tech University
+ */
+
+#include<iostream>
+#include "TrigScint/QIEInputPulse.h"
+#include<cmath>
+
+namespace trigscint {
+
+  void QIEInputPulse::AddPulse(float toff, float ampl){
+    toff_.push_back(toff);
+    ampl_.push_back(ampl);
+    npulses++;
```

Comment:
So npulses is just keeping track of how many elements are in toff and ampl? If so, why even have it when you can use the .size() method on each of the vectors above?

Reply:
Indeed, `npulses` is there to just keep the track of the number of pulses. Many of the methods of `QIEInputPulse` have for loops that loop over these vectors and I thought that storing the vector size explicitly could reduce the computation time by not having to evaluate `vector<>.size()` every time.

Reply:
The amount of time you are saving here is likely negligible given that you are incrementing the as you add to the vectors. I guess you could do something npulses = vector<>.size() once you are done filling the collections, but I doubt it makes much difference overall.